uefi-raw/src/protocol/firmware_volume.rs

```diff
@@ -0,0 +1,219 @@
+use crate::firmware_storage::FirmwareVolumeAttributes;
+use crate::protocol::block::Lba;
+use crate::{guid, Guid, Handle, PhysicalAddress, Status};
+use core::ffi::c_void;
+
+// EFI_FV_ATTRIBUTES
+bitflags::bitflags! {
+    #[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+    #[repr(transparent)]
+    pub struct FvAttributes: u64 {
+        const READ_DISABLE_CAP = 1 << 0;
+        const READ_ENABLE_CAP = 1 << 1;
+        const READ_STATUS = 1 << 2;
+
+        const WRITE_DISABLE_CAP = 1 << 3;
+        const WRITE_ENABLE_CAP = 1 << 4;
+        const WRITE_STATUS = 1 << 5;
+
+        const LOCK_CAP = 1 << 6;
+        const LOCK_STATUS = 1 << 7;
+        const WRITE_POLICY_RELIABLE = 1 << 8;
+        const READ_LOCK_CAP = 1 << 12;
+        const READ_LOCK_STATUS = 1 << 13;
+        const WRITE_LOCK_CAP = 1 << 14;
+        const WRITE_LOCK_STATUS = 1 << 15;
+
+        const ALIGNMENT = 0x1F << 16;
+        const ALIGNMENT_1 = 0x00 << 16;
+        const ALIGNMENT_2 = 0x01 << 16;
+        const ALIGNMENT_4 = 0x02 << 16;
+        const ALIGNMENT_8 = 0x03 << 16;
+        const ALIGNMENT_16 = 0x04 << 16;
+        const ALIGNMENT_32 = 0x05 << 16;
+        const ALIGNMENT_64 = 0x06 << 16;
+        const ALIGNMENT_128 = 0x07 << 16;
+        const ALIGNMENT_256 = 0x08 << 16;
+        const ALIGNMENT_512 = 0x09 << 16;
+        const ALIGNMENT_1K = 0x0A << 16;
+        const ALIGNMENT_2K = 0x0B << 16;
+        const ALIGNMENT_4K = 0x0C << 16;
+        const ALIGNMENT_8K = 0x0D << 16;
+        const ALIGNMENT_16K = 0x0E << 16;
+        const ALIGNMENT_32K = 0x0F << 16;
+        const ALIGNMENT_64K = 0x10 << 16;
+        const ALIGNMENT_128K = 0x11 << 16;
+        const ALIGNMENT_256K = 0x12 << 16;
+        const ALIGNMENT_512K = 0x13 << 16;
+        const ALIGNMENT_1M = 0x14 << 16;
+        const ALIGNMENT_2M = 0x15 << 16;
+        const ALIGNMENT_4M = 0x16 << 16;
+        const ALIGNMENT_8M = 0x17 << 16;
+        const ALIGNMENT_16M = 0x18 << 16;
+        const ALIGNMENT_32M = 0x19 << 16;
+        const ALIGNMENT_64M = 0x1A << 16;
+        const ALIGNMENT_128M = 0x1B << 16;
+        const ALIGNMENT_256M = 0x1C << 16;
+        const ALIGNMENT_512M = 0x1D << 16;
+        const ALIGNMENT_1G = 0x1E << 16;
+        const ALIGNMENT_2G = 0x1F << 16;
+    }
+}
+
+// EFI_FV_FILE_ATTRIBUTES
+bitflags::bitflags! {
+    #[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+    #[repr(transparent)]
+    pub struct FvFileAttributes: u32 {
+        const ALIGNMENT = 0x1F;
+        const FIXED = 1 << 8;
+        const MEMORY_MAPPED = 1 << 9;
+    }
+}
+
+// EFI_FV_WRITE_POLICY
+newtype_enum! {
+    pub enum FvWritePolicy: u32 => {
+        EFI_FV_UNRELIABLE_WRITE = 0,
+        EFI_FV_RELIABLE_WRITE = 1,
+    }
+}
+
+// EFI_FV_FILETYPE
+pub type FvFiletype = u8;
```

Comment:
This could add definitions, but I wasn't sure how to handle it because of the ranges like:

```
EFI_FV_FILETYPE_OEM_MIN 0xC0
EFI_FV_FILETYPE_OEM_MAX 0xDF
```